src/GHCJS/DOM/JSFFI/FormData.hs

```diff
@@ -0,0 +1,48 @@
+{-# LANGUAGE ForeignFunctionInterface, JavaScriptFFI #-}
+module GHCJS.DOM.JSFFI.FormData (
+    module Generated
+  , js_append
+  , append
+  , appendBlob
+  , appendFile
+) where
+
+import Prelude hiding (appendFile)
+import Control.Monad.IO.Class (MonadIO(..))
+
+import GHCJS.Types (JSVal, JSString)
+import GHCJS.Marshal.Internal (PToJSVal(..))
+import GHCJS.Foreign (jsUndefined)
+import GHCJS.DOM.Types
+
+import GHCJS.DOM.JSFFI.Generated.FormData as Generated hiding (js_append, append)
+
+
+maybeToOptional :: PToJSVal a => Maybe a -> JSVal
+maybeToOptional (Just a) = pToJSVal a
+maybeToOptional Nothing = jsUndefined
+
+
+foreign import javascript unsafe "$1[\"append\"]($2, $3, $4)"
+        js_append :: FormData -> JSString -> JSVal -> JSVal -> IO ()
+
+-- | <https://developer.mozilla.org/en-US/docs/Web/API/FormData/append Mozilla FormData.append documentation>
+append
+  :: (MonadIO m, ToJSString name, ToJSString str)
+  => FormData -> name -> str -> m ()
+append self name str
+  = liftIO $ js_append self (toJSString name) (pToJSVal str) jsUndefined
```

Comment:
I just did some testing in Chrome, and it seems that passing undefined is *not* equivalent to calling with fewer arguments, unfortunately.  See http://ryantrinkle.s3.amazonaws.com/screenshot_on_2016-06-04-22:01:34_434x184_T_shjTeK5jHqSuyCySM_mEQ5Pk4vbxkqqhh0sOtw4pE=.png